agreement/models/agreement.py

```diff
@@ -346,7 +346,7 @@ def onchange_copyvalue(self):
             self.sub_model_object_field_id = False
         if self.field_id and self.field_id.relation:
             self.sub_object_id = self.env['ir.model'].search(
-                    [('model', '=', self.field_id.relation)])[0]
+                [('model', '=', self.field_id.relation)])[0]
             if self.sub_model_object_field_id:
```

Comment:
```suggestion
        if self.sub_model_object_field_id:
            self.copyvalue = "${object.%s.%s or %s}" %\
                                 (self.field_id.name,
                                  self.sub_model_object_field_id.name,
                                  self.default_value or '\'\'')

```
This works for me locally.  I don't seem to be able to get a working runbot instance (https://runbot.odoo-community.org/runbot/repo/github-com-oca-contract-110)